12/umbraco-cms/content-delivery-api.md

```diff
@@ -0,0 +1,413 @@
+---
+
+# Content Delivery API
+
+## Overview
+
+With its several extension points, this new API delivers headless capabilities built directly into Umbraco. It allows you to retrieve your content items in a JSON format and lets you preset them in different channels, using your preferred technology stack. This feature preserves the friendly editing experience of Umbraco, while also ensuring a performant delivery of content, even in headless scenarios.
+
+## Getting Started
+
+When upgrading an existing project to version 12, you would need to opt-in explicitly for using the Delivery API. Below you will find the steps you need to take in order to configure it for your Umbraco project. If you start with a fresh Umbraco 12 installation, the Delivery API will be enabled by default and therefore you can skip straight to the [Additional configuration](content-delivery-api.md#additional-configuration) section.
+
+### Register the Content Delivery API dependencies
+
+In the `Startup.cs` file, register the API dependencies in the `ConfigureServices` by adding `.AddDeliveryApi()`:
+
+```csharp
+public void ConfigureServices(IServiceCollection services)
+{
+    services.AddUmbraco(_env, _config)
+        .AddBackOffice()             
+        .AddWebsite()
+        // Register all Delivery API dependencies
+        .AddDeliveryApi()
+        .AddComposers()
+        .Build();
+}
+```
+
+### Enable the Content Delivery API
+
+Add the `DeliveryApi` configuration section in `appsettings.json` and set the value of the `Enabled` key to `true`:
+
+{% code title="appsettings.json" %}
+```json
+{
+    "Umbraco": {
+        "CMS": {
+            "DeliveryApi": {
+                "Enabled": true
+            }
+        }
+    }
+}
+```
+{% endcode %}
+
+### Additional configuration
+
+Once the Delivery API has been configured on your project, you need to be aware that all your content will be made available to the public. However, there are a few additional configuration options that you can use to restrict access to the Delivery API endpoints and limit the content that is returned.
+
+{% code title="appsettings.json" %}
+```json
+{
+    "Umbraco": {
+        "CMS": {
+            "DeliveryApi": {
+                "Enabled": true,
+                "PublicAccess": true,
+                "ApiKey": "my-api-key",
+                "DisallowedContentTypeAliases": ["alias1", "alias2", "alias3"]
+            }
+        }
+    }
+}
+```
+{% endcode %}
+
+* `Umbraco.CMS.DeliveryApi.PublicAccess` determines whether the Delivery API (_if enabled_) should be publicly accessible or if access should require an API key.
+* `Umbraco.CMS.DeliveryApi.ApiKey` specifies what API key to use for authorizing access to the API when public access is disabled. This setting is also used for managing access to draft content.
+* `Umbraco.CMS.DeliveryApi.DisallowedContentTypeAliases` contains the aliases of the content types that would never be exposed through the Delivery API.
+
```

Comment:
We need to document the `"RichTextOutputAsJson"` setting too. At this point we probably don't need to dive too deep into the actual functionality, so maybe something like

```suggestion
* `Umbraco:CMS:DeliveryApi:RichTextOutputAsJson` enables outputting rich text content as JSON rather than the default HTML output. JSON can be a preferred format in many scenarios, not least because it supports routing of internal links better than HTML does.
```

Reply:
I decided to leave it out of this section because it is meant to cover "_additional configuration options that you can use to **restrict access** to the Delivery API endpoints and **limit the content** that is returned_" (_as per the introduction to the section_). We can either write a transition sentence after the three "original" settings and then mention that it also exists or leave it for another article where demonstrate how to interact with the endpoints 🤔 

What would you prefer?

Reply:
Yeah I gathered as much. But the setting will be there by default (as false) for a newly created V12 project, so we should document it while we're documenting the others. A transition sentence would be great 👍

Reply:
Alright, I will add it then 👍